kernel/kernel-api/src/main/java/io/delta/kernel/internal/DeltaErrors.java

```diff
@@ -274,6 +275,34 @@ public static KernelException invalidConfigurationValueException(
     return new InvalidConfigurationValueException(key, value, helpMessage);
   }
 
+  public static KernelException domainMetadataUnsupported() {
```

Comment:
I think this is a bit ambiguous as to whether it's for reads or writes. I had assumed this means we are reading an invalid table, but I think it's an error on write?

Maybe clarify with something like "cannot commit DomainMetadata action(s) due to insufficient table protocol. .....[etc]"?

Reply:
Yes, this is an error on write. The error message was indeed a bit ambiguous. I've updated it to
```
Cannot commit DomainMetadata action(s) because the feature 'domainMetadata' is not supported on this table.
```
Hope it is clearer now. Thanks for pointing this out!